barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -259,7 +259,8 @@ HonkProof ClientIVC::construct_and_prove_hiding_circuit()
     // circuit. So, these have to be preserved as public inputs to the hiding circuit (and, subsequently, as public
     // inputs to the tube circuit) which are intermediate stages.
     // TODO(https://github.com/AztecProtocol/barretenberg/issues/1048): link these properly, likely insecure
-    auto num_public_inputs = static_cast<uint32_t>(static_cast<uint256_t>(fold_proof[PUBLIC_INPUTS_SIZE_INDEX]));
+    auto num_public_inputs =
+        static_cast<uint32_t>(static_cast<uint256_t>(verification_queue[0].honk_verification_key->num_public_inputs));
```

Comment:
This should be equivalent to the following if you prefer:
```suggestion
        static_cast<uint32_t>(static_cast<uint256_t>(honk_vk->num_public_inputs));
```